File: indexter-games/src/main/java/com/github/ykrasik/indexter/games/datamodel/Library.java
package com.github.ykrasik.indexter.games.datamodel;

import com.google.common.base.MoreObjects;

import java.nio.file.Path;
import java.util.Objects;

/**
 * @author Yevgeny Krasik
 */
public class Library {
    private final String name;
    private final Path path;
    private final GamePlatform platform;

    public Library(String name, Path path, GamePlatform platform) {
        this.name = Objects.requireNonNull(name);
        this.path = Objects.requireNonNull(path);
        this.platform = Objects.requireNonNull(platform);
    }

    public String getName() {
        return name;
    }

    public Path getPath() {
        return path;
    }

    public GamePlatform getPlatform() {
        return platform;
    }

    @Override
    public String toString() {
        return MoreObjects.toStringHelper(this)
            .add("name", name)
            .add("path", path)
            .add("platform", platform)
            .toString();
    }
}

File: gamedex-common/src/main/kotlin/com/gitlab/ykrasik/gamedex/datamodel/Library.kt
package com.gitlab.ykrasik.gamedex.datamodel

import java.io.File

/**
 * User: ykrasik
 * Date: 25/05/2016
 * Time: 11:29
 */
data class Library(
    val id: Int,
    val path: File,
    val data: LibraryData
) : Comparable<Library> {

    val name: String get() = data.name
    val platform: GamePlatform get() = data.platform

    override fun compareTo(other: Library) = name.compareTo(other.name)
}

data class LibraryData(
    val platform: GamePlatform,
    val name: String
)